### PR TITLE
app-editors/visual-studio-code: New Ebuild

### PR DESCRIPTION
**Pull request by oniGino**
Bug: https://bugs.gentoo.org/675742

* New Ebuild for Visual Studio Code (Pre Built Binary from Microsoft)

Signed-off-by: Gino McCarty <onigino@protonmail.com>

### CODE REVIEW

**Comment by Gaelan** [comment]
The license here is wrong—the VS Code source is MIT, but the [official binaries have a EULA](https://code.visualstudio.com/license).

**Comment by oniGino** [comment]
> The license here is wrong—the VS Code source is MIT, but the [official binaries have a EULA](https://code.visualstudio.com/license).

OK, inserted a text version of the license

**Comment by oniGino** [comment]
any way to get this reviewed please? its been open forever

**Comment by rossbridger** [comment]
I think the package name should be _visual-studio-code-bin_ since it's a binary build.

**Comment by oniGino** [comment]
> I think the package name should be _visual-studio-code-bin_ since it's a binary build.

plenty of precedence for this, one popular example www-client/google-chrome
but if maintainers wants it changed I can

**Comment by rossbridger** [comment]
Since building vscode directly from upstream source is also possible, there should be a name difference in case new package for building from source will also be available in the future, like dev-java/openjdk and dev-java/openjdk-bin.

**Comment by rossbridger** [comment]
ping

**Comment by Gaelan** [comment]
@rossbridger it's more of a chrome/chromium situation: There's "Visual Studio Code" (analogous to Google Chrome), which is a binaries distributed by MS under a proprietary license with some extra features, and the awkwardly-named "Code — OSS" (analogous to Chromium) which is what you get if you compile the open source project. So you can't compile "Visual Studio Code" from source per se

**Comment by rossbridger** [comment]
@Gaelan I guess we can possibly rename the source build to something like visual-studio-code-oss or something, but it might still get confusing given Microsoft actually named its source respository [microsoft/vscode](https://github.com/microsoft/vscode), not vscode-oss or something else.

**Comment by oniGino** [comment]
ok, i renamed it and bumped to latest release 1.47, thanks for the feedback.

**Comment by rossbridger** [comment]
ping @gentoo/proxy-maint

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (Azulinho) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:39]: this should be:

```fperms +x "/opt/${PN}/bin/code"```

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (Azulinho) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:34]: this should be

```dosym "../../opt/${PN}/bin/code" "usr/bin/${PN}"```

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (thesamesam) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:10]:
```suggestion
SRC_URI="amd64? ( https://update.code.visualstudio.com/${PV}/linux-x64/stable -> ${P}.tar.gz )"
```

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (thesamesam) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:16]:
```suggestion
KEYWORDS="-* ~amd64"
```
because it's a binary and can't work [elsewhere](https://devmanual.gentoo.org/keywording/index.html)

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (thesamesam) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:17]: We can [drop](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-620007.3) empty IUSE.

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (thesamesam) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:22]: Sort?

File: app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild
Comment (thesamesam) [inline app-editors/visual-studio-code-bin/visual-studio-code-bin-1.47.3.ebuild:27]: We should set S in global scope, right below SRC_URI or right above the first phase definition. So we can get rid of pkg_setup here.

**Review by thesamesam — CHANGES_REQUESTED** [review]

**Comment by rossbridger** [comment]
@oniGino has been unresponsive for quite a while, certainly not on track for maintaining a frequently updated package like vscode. I think we can close this PR now.

**Comment by oniGino** [comment]
> @oniGino has been unresponsive for quite a while, certainly not on track for maintaining a frequently updated package like vscode. I think we can close this PR now.

I have made every change requested for this PR and am waiting for review/merge. 

What else do you want?

**Comment by rossbridger** [comment]
> I have made every change requested for this PR and am waiting for review/merge.
You have already received reviews in the past two months, but you haven't resolved any of them.

**Pull request closed without merging**
